modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -89,32 +89,24 @@ class Instrument_Manager extends \NDB_Menu_Filter
                 array('testname' => $_FILES['install_file']['name'])
             );
             if ($exists <= 0) {
-                $db_config = $config->getSetting('database');
                 exec(
                     'php generate_tables_sql_and_testNames.php < '
-                    . escapeshellarg($new_file)
+                    . escapeshellarg($new_file),
+                    $error_message
                 );
-
-                $instrument = \NDB_BVL_Instrument::factory(
-                    $instname,
-                    '',
-                    ''
-                );
-                if (!isset($db_config['quatUser'], $db_config['quatPassword'])) {
+                $error_message = implode($error_message);
```

Comment:
This looks like array-string conversion and should probably be avoided.

This could probably be better written as
```php
$output = shell_exec(
    'php generate_tables_sql_and_testNames.php < '
    . escapeshellarg($new_file)
);
```

Note that both `exec` and `shell_exec` return all output from the external command. You should be sure that `generate_tables_sql_and_testNames.php` is silent on success if you are going to treat any non-empty string as an error message.